src/main/java/org/jabref/gui/search/GlobalSearchResultDialog.java

```diff
@@ -76,5 +77,15 @@ private void initialize() {
                     ? IconTheme.JabRefIcons.KEEP_ON_TOP.getGraphicNode()
                     : IconTheme.JabRefIcons.KEEP_ON_TOP_OFF.getGraphicNode());
         });
+
+        getDialogPane().getScene().getWindow().addEventHandler(WindowEvent.WINDOW_SHOWN, event -> {
+            getDialogPane().setPrefHeight(preferencesService.getSearchPreferences().getSearchWindowHeight());
+            getDialogPane().setPrefWidth(preferencesService.getSearchPreferences().getSearchWindowWidth());
+        });
+
+        getDialogPane().getScene().getWindow().addEventHandler(WindowEvent.WINDOW_HIDDEN, event -> {
+            preferencesService.getSearchPreferences().setSearchWindowHeight((int) getHeight());
+            preferencesService.getSearchPreferences().setSearchWindowWidth((int) getWidth());
+        });
```

Comment:
```suggestion
        setWidth(preferencesService.getSearchPreferences().getSearchWindowWidth());
        setHeight(preferencesService.getSearchPreferences().getSearchWindowHeight());

        getDialogPane().getScene().getWindow().addEventHandler(WindowEvent.WINDOW_HIDDEN, event -> {
            preferencesService.getSearchPreferences().setSearchWindowHeight(getHeight());
            preferencesService.getSearchPreferences().setSearchWindowWidth(getWidth());
        });
```

Reply:
Height and width should be stored as doubles as they are kept by javafx as doubles to avoid casting.

Reply:
For some reason, window size is not retained when doing it that way.

Reply:
First of all, the `setWidth()` and `setHeight()` do not change the size of the dialog window. This has to be done with `getDialogPane().setPrefHeight()` and `getDialogPane().setPrefWidth()`. 

Also, when setting the size of the window as in the suggestion of @calixtus, it does not actually change the window size. It only works when adding an event handler to the window. This event handler sets the size of the window when the `WINDOW_SHOWN` event occurs, hence when the window is opened. 

My current solution does correctly retain the size of the window.

Reply:
hm... was experimenting a little bit on my machine. must have missed something. good to know you could clear that.

Reply:
Thanks for the info! I'm fine with the changes now